apps/explorer/src/app/routes/blocks/home/index.tsx

```diff
@@ -1,29 +1,95 @@
+import { useState } from 'react';
 import { DATA_SOURCES } from '../../../config';
-import useFetch from '../../../hooks/use-fetch';
-import type { TendermintBlockchainResponse } from '../tendermint-blockchain-response';
+import type {
+  BlockMeta,
+  TendermintBlockchainResponse,
+} from '../tendermint-blockchain-response';
 import { RouteTitle } from '../../../components/route-title';
-import { RenderFetched } from '../../../components/render-fetched';
-import { BlocksData, BlocksRefetch } from '../../../components/blocks';
+import { BlocksRefetch } from '../../../components/blocks';
+import { BlocksInfiniteList } from '../../../components/blocks/blocks-infinite-list';
 import { JumpToBlock } from '../../../components/jump-to-block';
-import { t } from '@vegaprotocol/react-helpers';
+import { t, useFetch } from '@vegaprotocol/react-helpers';
+
+// This constant should only be changed if Tendermint API changes the max blocks returned
+const TM_BLOCKS_PER_REQUEST = 20;
 
 const Blocks = () => {
+  const [hasMoreBlocks, setHasMoreBlocks] = useState(true);
```

Comment:
suggestion (non blocking): Quite a lot of use states here, might be nice to refactor into a reducer or put everything into a single object.

Reply:
Good plan